depends/toolchain.cmake.in

```diff
@@ -0,0 +1,195 @@
+    endforeach()
+  else()
+    set(${compiler} ${${env_compiler}})
+  endif()
+  set(${launcher} ${${launcher}} PARENT_SCOPE)
+  set(${compiler} ${${compiler}} PARENT_SCOPE)
+endfunction()
+
+if(NOT DEFINED CMAKE_C_FLAGS_INIT)
+  set(CMAKE_C_FLAGS_INIT "@CFLAGS@")
+endif()
+if(NOT DEFINED CMAKE_C_FLAGS_RELWITHDEBINFO_INIT)
+  set(CMAKE_C_FLAGS_RELWITHDEBINFO_INIT "@CFLAGS_RELEASE@")
+endif()
+if(NOT DEFINED CMAKE_C_FLAGS_DEBUG_INIT)
+  set(CMAKE_C_FLAGS_DEBUG_INIT "@CFLAGS_DEBUG@")
+endif()
+
+if(NOT DEFINED CMAKE_C_COMPILER)
+  set(DEPENDS_C_COMPILER_WITH_LAUNCHER @CC@)
+  split_compiler_launcher(DEPENDS_C_COMPILER_WITH_LAUNCHER CMAKE_C_COMPILER_LAUNCHER CMAKE_C_COMPILER)
+  set(CMAKE_C_LINKER_LAUNCHER ${CMAKE_C_COMPILER_LAUNCHER})
+endif()
+
+if(NOT DEFINED CMAKE_CXX_FLAGS_INIT)
+  set(CMAKE_CXX_FLAGS_INIT "@CXXFLAGS@")
+  set(CMAKE_OBJCXX_FLAGS_INIT "@CXXFLAGS@")
+endif()
+if(NOT DEFINED CMAKE_CXX_FLAGS_RELWITHDEBINFO_INIT)
+  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO_INIT "@CXXFLAGS_RELEASE@")
+  set(CMAKE_OBJCXX_FLAGS_RELWITHDEBINFO_INIT "@CXXFLAGS_RELEASE@")
+endif()
+if(NOT DEFINED CMAKE_CXX_FLAGS_DEBUG_INIT)
+  set(CMAKE_CXX_FLAGS_DEBUG_INIT "@CXXFLAGS_DEBUG@")
+  set(CMAKE_OBJCXX_FLAGS_DEBUG_INIT "@CXXFLAGS_DEBUG@")
+endif()
+
+if(NOT DEFINED CMAKE_CXX_COMPILER)
+  set(DEPENDS_CXX_COMPILER_WITH_LAUNCHER @CXX@)
+  split_compiler_launcher(DEPENDS_CXX_COMPILER_WITH_LAUNCHER CMAKE_CXX_COMPILER_LAUNCHER CMAKE_CXX_COMPILER)
+  set(CMAKE_CXX_LINKER_LAUNCHER ${CMAKE_CXX_COMPILER_LAUNCHER})
+
+  set(CMAKE_OBJCXX_COMPILER ${CMAKE_CXX_COMPILER})
+  set(CMAKE_OBJCXX_COMPILER_LAUNCHER ${CMAKE_CXX_COMPILER_LAUNCHER})
+  set(CMAKE_OBJCXX_LINKER_LAUNCHER ${CMAKE_CXX_LINKER_LAUNCHER})
+endif()
+
+# The DEPENDS_COMPILE_DEFINITIONS* variables are to be treated as lists.
+set(DEPENDS_COMPILE_DEFINITIONS @CPPFLAGS@)
+set(DEPENDS_COMPILE_DEFINITIONS_RELWITHDEBINFO @CPPFLAGS_RELEASE@)
+set(DEPENDS_COMPILE_DEFINITIONS_DEBUG @CPPFLAGS_DEBUG@)
+
+if(NOT DEFINED CMAKE_EXE_LINKER_FLAGS_INIT)
+  set(CMAKE_EXE_LINKER_FLAGS_INIT "@LDFLAGS@")
+endif()
+if(NOT DEFINED CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO_INIT)
+  set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO_INIT "@LDFLAGS_RELEASE@")
+endif()
+if(NOT DEFINED CMAKE_EXE_LINKER_FLAGS_DEBUG_INIT)
+  set(CMAKE_EXE_LINKER_FLAGS_DEBUG_INIT "@LDFLAGS_DEBUG@")
+endif()
+
+set(CMAKE_AR "@AR@")
+set(CMAKE_RANLIB "@RANLIB@")
+set(CMAKE_STRIP "@STRIP@")
+set(CMAKE_OBJCOPY "@OBJCOPY@")
+set(CMAKE_OBJDUMP "@OBJDUMP@")
+
+# Using our own built dependencies should not be
+# affected by a potentially random environment.
+set(CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH OFF)
+
+set(CMAKE_FIND_ROOT_PATH "@depends_prefix@")
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+set(QT_TRANSLATIONS_DIR "@depends_prefix@/translations")
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT CMAKE_HOST_APPLE)
```

Comment:
Is the intent to handle any cross-compilation to Apple platforms from non-Apple hosts, or specifically cross-compilation to macOS?
Is this the same as
```
CMAKE_CROSSCOMPILING AND NOT CMAKE_HOST_APPLE
```
?

Reply:
> Is the intent to handle any cross-compilation to Apple platforms from non-Apple hosts, or specifically cross-compilation to macOS?

The former.

> Is this the same as
> 
> ```
> CMAKE_CROSSCOMPILING AND NOT CMAKE_HOST_APPLE
> ```
> 
> ?

In the context of a toolchain file, it is not. The `CMAKE_CROSSCOMPILING` variable is well-defined after a `project()` call, but not within a toolchain file.

Reply:
Thanks